docker/token/token.go

```diff
@@ -0,0 +1,54 @@
+package token
+
+import (
+	"context"
+	"github.com/Portshift/klar/docker/token/ecr"
+	"github.com/Portshift/klar/docker/token/gcr"
+	"github.com/Portshift/klar/docker/token/secret"
+	"github.com/containers/image/v5/docker/reference"
+	log "github.com/sirupsen/logrus"
+)
+
+type CredExtractor struct {
+	extractors []Extractor
+}
+
+type Extractor interface {
+	// Prints the name of the extractor
+	Name() string
+	// Returns true if extractor is matched for extracting credentials for the given image
```

Comment:
Returns true if extractor is supported for extracting credentials for the given image